caom2-meta-server/src/main/java/ca/nrc/cadc/caom2/meta/MetaQueryRunner22.java

```diff
@@ -0,0 +1,345 @@
+                runID = job.getRunID();
+            CaomTapQuery query = new CaomTapQuery(tapID, runID);
+            Observation obs = query.performQuery(uri);
+            log.warn("found: " + obs);
+            
+            if (obs == null)
+            {
+                throw new ObservationNotFoundException(uri);
+            }
+                
+            syncOutput.setResponseCode(HttpURLConnection.HTTP_OK);
+            syncOutput.setHeader("Content-Type", format);
+            
+            if (JSON_FORMAT.equals(format))
+            {
+                JsonWriter writer = new JsonWriter(true, XmlConstants.CAOM2_2_NAMESPACE);
+                writer.write(obs, syncOutput.getOutputStream());
+            }
+            else
+            {
+                //RegistryClient reg = new RegistryClient();
+                //URL metaURL = reg.getServiceURL(metaID, Standards.CAOM2_OBS_20, AuthMethod.ANON);
+                //String styleSheetURL = metaURL.toExternalForm().replace("/meta", "/caom2_summary.xslt");
+                ObservationWriter writer = new ObservationWriter("caom2", XmlConstants.CAOM2_2_NAMESPACE, false);
+                //writer.setStylesheetURL(styleSheetURL);
+                writer.write(obs, syncOutput.getOutputStream());
+            }
+            
+            // set final phase, only sync so no results
+            log.debug(job.getID() + ": EXECUTING -> COMPLETED...");
+            ep = jobUpdater.setPhase(job.getID(), ExecutionPhase.EXECUTING, ExecutionPhase.COMPLETED, new Date());
+            if ( !ExecutionPhase.COMPLETED.equals(ep) )
+            {
+                ep = jobUpdater.getPhase(job.getID());
+                log.debug(job.getID() + ": EXECUTING -> COMPLETED [FAILED], phase was " + ep);
+                logInfo.setSuccess(false);
+                logInfo.setMessage("Could not set job phase to completed.");
+                return;
+            }
+            log.debug(job.getID() + ": EXECUTING -> COMPLETED [OK]");
+
+        }
+        catch(CertificateException ex)
+        {
+            sendError(ex, "permission denied -- reason: invalid proxy certificate", 403);
+        }
+        catch(IllegalArgumentException ex)
+        {
+            sendError(ex, ex.getMessage(), 400);
+        }
+        catch(ObservationNotFoundException ex)
+        {
+            sendError(ex, ex.getMessage(), 404);
+        }
+        catch(UnsupportedOperationException ex)
+        {
+            sendError(ex, "unsupported operation: " + ex.getMessage(), 400);
+        }
+        catch(TransientFault ex)
+        {
+            sendError(ex, ex.getMessage(), ex.getResponseCode());
+        }
+        catch(Throwable t)
+        {
+            if ( ThrowableUtil.isACause(t, InterruptedException.class) )
+            {
+                try
+                {
+                    ep = jobUpdater.setPhase(job.getID(), ExecutionPhase.QUEUED, ExecutionPhase.EXECUTING, new Date());
```

Comment:
this is correct. The UWS servlet puts it from pending to queued and job puts it to executing